Documentation/CRDs/Block-Storage/ceph-block-pool-crd.md

```diff
@@ -232,6 +232,12 @@ stretched) then you will have 2 replicas per datacenter where each replica ends
     !!! note
         A value of 0 disables the quota.
 
+* `pgs`: Set a custom PG count for the pool. By default, the PGs will be managed by the Ceph PG autoscaler. See the [Ceph documentation](https://docs.ceph.com/en/latest/rados/operations/pools/#create-a-pool) for more info.
+* `autoscaleMode`: set autoscale mode to pool. See the [ceph documentation](https://docs.ceph.com/en/latest/rados/operations/placement-groups/) for more info.
+    * `off`: Disable autoscaling for this pool. It is up to the administrator to choose an appropriate pgp_num for each pool. Please refer to [Choosing the number of Placement Groups](https://docs.ceph.com/en/latest/rados/operations/placement-groups/#choosing-number-of-placement-groups) for more information.
+    * `on`: Enable automated adjustments of the PG count for the given pool.
+    * `warn`: Raise health alerts when the PG count should be adjusted.
+
```

Comment:
Should we document that setting this to `""` (or leaving it unset) reverts to Ceph's default behavior?

Reply:
no, if  the PlacementGroupCount > 0, we should set the autoscalemode to warn or off in default. if PlacementGroupCount not set, the default value of the autoscalemode  should be on.

Reply:
But Ceph should have its own default behavior. Why can we not just let ceph handle its own defaults.

Reply:
If user set the pg_num means want to disable the autoscalemode, otherwise it doesn't make any sense.

Reply:
You are correct, but seemed to be missing my intent. Ceph often has its own behavior relating to configurations, and it is often best for Rook to rely on Ceph behaviors when they exist. In this case, after inspecting Ceph's C++ code, I did not find any Ceph behaviors that already accomplish what is being implemented here, so it is appropriate for Rook to do so.